src/controller/CHIPDevice.h

```diff
@@ -71,6 +73,16 @@ class DLL_EXPORT Device
      */
     CHIP_ERROR SendMessage(System::PacketBuffer * message);
 
+    /**
+     * @brief
+     *   Send the provided message to the device
+     *
+     * @param[in] message   The message to be sent.
+     *
+     * @return CHIP_ERROR   CHIP_NO_ERROR on success, or corresponding error
+     */
+    CHIP_ERROR SendMessage(System::PacketBufferHandle message);
```

Comment:
does this message contain the endpoint and cluster id?  I'd prefer to see something like:

`SendMessage(<message params>, Callback::Callback<> & onResponse)`

Reply:
The message contains the ZCL encoded packet. The packet is encoded by the cluster class using the ZAP generated code. (not sure if it answer the question).

Reply:
The response processing still needs to be integrated to the code. So maybe we can follow your suggestion once that comes in.